covenant/covenant.go

```diff
@@ -409,6 +447,25 @@ func (ce *CovenantEmulator) covenantSigSubmissionLoop() {
 
 }
 
+func (ce *CovenantEmulator) metricsUpdateLoop() {
+	defer ce.wg.Done()
+
+	interval := ce.config.Metrics.UpdateInterval
+	ce.logger.Info("starting metrics update loop",
+		zap.Float64("interval seconds", interval.Seconds()))
+
+	for {
+		metricsTimeKeeper.UpdatePrometheusMetrics()
+
+		select {
+		case <-time.After(interval):
```

Comment:
Can this be done with a ticker as follows?


```suggestion
		select {
		case <-ticker.C:
            metricsTimeKeeper.UpdatePrometheusMetrics()
```